src/routes/common/interceptors/route-logger.interceptor.ts

```diff
@@ -1,29 +1,90 @@
 import {
   CallHandler,
   ExecutionContext,
+  HttpException,
+  HttpStatus,
   Injectable,
   NestInterceptor,
 } from '@nestjs/common';
 import { ILoggingService, LoggingService } from '../logging/logging.interface';
 import { Inject } from '@nestjs/common/decorators';
+import { Observable, tap } from 'rxjs';
 
 /**
  * The {@link RouteLoggerInterceptor} is an interceptor that logs the requests
  * that target a specific route.
  *
  * Since this is an interceptor we have access to the respective {@link ExecutionContext}
+ * and have, therefore, more data regarding which route handled this request
  * See https://docs.nestjs.com/fundamentals/execution-context
+ *
+ * Note: this interceptor is triggered if there is a matching route. Therefore,
+ * if a request is made to a non-existing route (resulting in a 404) this interceptor
+ * does not log such event.
  */
 @Injectable()
 export class RouteLoggerInterceptor implements NestInterceptor {
+  private static LOG_FORMAT = '[==>] %s %s %d';
```

Comment:
nitpick: should we extract this to a common place, since the same format is used in `NotFoundLoggerMiddleware`? (i.e.: does it make sense to tie both loggers to a common format?)

Reply:
> nitpick: should we extract this to a common place, since the same format is used in `NotFoundLoggerMiddleware`? (i.e.: does it make sense to tie both loggers to a common format?)

Yeah that's a good point! We could move to a shared module however I'm in the process of reviewing the format itself so my suggestion is to either tackle it in a different PR or change the format completely (depending on how valuable we find the new format that is!)